subsys/net/lib/download_client/src/download_client.c

```diff
@@ -742,6 +756,9 @@ int download_client_disconnect(struct download_client *const client)
 	}
 
 	client->fd = -1;
+	client->close_when_done = false;
+	client->host = NULL;
```

Comment:
I got an assert on http.c:57 because of this. We have to be carefull changing this variables from multiple threads.

Reply:
This looks like we actually need to add mutex to protect the content of client structure.